src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java

```diff
@@ -181,6 +182,25 @@ public long countObservationsByInstance(final Integer studyId, final Integer dat
 	@Override
 	public MeasurementVariable addDatasetVariable(final Integer studyId, final Integer datasetId, final DatasetVariable datasetVariable) {
 		this.studyValidator.validate(studyId, true);
+
+		return this.getMeasurementVariable(studyId, datasetId, datasetVariable);
+
+	}
+
+	@Override
+	public void addDatasetVariables(final Integer studyId, final List<DatasetVariable> datasetVariables) {
+		this.studyValidator.validate(studyId, true);
+
+		final Integer datasetId = this.getDatasets(
```

Comment:
Make the datasetId as a method parameter so that this can be reused with other dataset.